rsconnect_jupyter/static/connect.js

```diff
@@ -1037,6 +1073,78 @@ define([
       });
   }
 
+  function onCreateManifestClicked() {
+    // This function will be passed (env, event) in the first two
+    // positional slots. We're not using them.
+
+    // lazily load the config when clicked since Jupyter's init
+    // function is racy w.r.t. loading of notebook metadata
+    if (!config) {
+      config = new RSConnect(debug);
+      window.RSConnect = config;
+    }
+
+    closeMenu();
+
+    var dialog = Dialog.modal({
+      // pass the existing keyboard manager so all shortcuts are disabled while
+      // modal is active
+      keyboard_manager: Jupyter.notebook.keyboard_manager,
+
+      title: 'Create Manifest',
+      body: [
+        '<p>',
+        '    Click Create Manifest to create the manifest.json and requirements.txt',
+        '    files needed for publishing to RStudio Connect via git.',
+        '    Note: requirements.txt will be created only if it does not already exist.',
+        '</p>',
```

Comment:
The note should probably be a second paragraph.

"Create Manifest" should probably be quoted to isolate it from the rest of the verbiage.

This seems like a pretty good summary of what's going on but I couldn't help but feel like I didn't understand why there was a dialog here. Perhaps we could explain to the user what the consequences of doing this if they don't actually need it are?

Reply:
I'd be interested to see a sample of manifest/requirements generation failing. I can reproduce a success locally, but I'm not sure I understand what failure looks like.

Success 1:

![Screen Shot 2019-10-06 at 3 39 51 PM](https://user-images.githubusercontent.com/1735264/66274734-93213c00-e84f-11e9-89e5-9566c669254a.png)

Shamelessly spamming the "Create Manifest" button:

![Screen Shot 2019-10-06 at 3 40 18 PM](https://user-images.githubusercontent.com/1735264/66274742-a46a4880-e84f-11e9-8219-45de1f69d05c.png)

Reply:
This is the expected behavior if you click the button a second time. Seems like we need some feedback that requirements.txt already existed and was not overwritten.